resen/Resen.py

```diff
@@ -832,19 +834,32 @@ def update_bucket_statuses(self):
             self.save_config()
 
 
+    def update_core_list(self):
+
+        core_list_url = 'https://raw.githubusercontent.com/EarthCubeInGeo/resen-core/master/cores.json'
+        core_filename = os.path.join(self.resen_root_dir,'cores','cores.json')
+
+        r = requests.get(core_list_url)
+        with open(core_filename, 'wb') as f:
+            f.write(r.content)
+
+
     def __get_valid_cores(self):
-        # TODO: download json file from resen-core github repo
-        #       and if that fails, fallback to hardcoded list
-        return [{"version":"2019.1.0","repo":"resen-core","org":"earthcubeingeo",
-                 "image_id":'sha256:5300c6652851f35d2fabf866491143f471a7e121998fba27a8dff6b3c064af35',
-                 "repodigest":'sha256:a8ff4a65aa6fee6b63f52290c661501f6de5bf4c1f05202ac8823583eaad4296'},
-                {"version":"2020.1.0","repo":"resen-core","org":"earthcubeingeo",
-                 "image_id":'sha256:b1f1c9013924c95f678a0aa7403e343cc2ee103f438b1a237193f091170ba077',
-                 "repodigest":'sha256:7bf4e28cf06e40b0e12cb0474232d6d3b520ec1a8c2d361d83fc0cb97083989c'},
-               ]
-    
-    
+        # define core list directory
+        core_dir = os.path.join(self.resen_root_dir,'cores')
+
+        # If core directory does not exist, create it and download the default core list file
+        if not os.path.exists(core_dir):
+            os.makedirs(core_dir)
+            self.update_core_list()
+
+        # for each JSON file in core directory, read in list of cores
+        cores = []
+        for fn in os.listdir(core_dir):
```

Comment:
Not a big deal right now, but this could have side effects:
1) doesn't check validity of JSON (do dictionaries have `version`, `image_id`, etc?
2) could fail if problem reading files

Something to keep in mind.

Reply:
Good point, particularly if we're going to have users add their own JSON files.  I've added exception handling if the JSON parsing fails (not a valid JSON file), but maybe we can hold off on checking that the dictionaries are correct until we decide exactly how the user interface for adding new cores is going to work (next release).